odxtools/dynamiclengthfield.py

```diff
@@ -107,11 +108,11 @@ def decode_from_pdu(self, decode_state: DecodeState) -> ParameterValue:
             odxraise(
                 f"Number of items specified by a dynamic length field {self.short_name} "
                 f"must be positive (is: {n})", DecodeError)
-        else:
-            decode_state.cursor_byte_position = decode_state.origin_byte_position + self.offset
-            result: List[ParameterValue] = []
-            for _ in range(n):
-                result.append(self.structure.decode_from_pdu(decode_state))
+            n = 0
+
+        decode_state.cursor_byte_position = decode_state.origin_byte_position + self.offset
+        for _ in range(n):
+            result.append(self.structure.decode_from_pdu(decode_state))
```

Comment:
Why is there changes for dynamic length field here?

Reply:
because I've took the dynamic length field code as a template and I noticed that it was hard to read. (at runtime the new code should be equivalent to the old one.)